.github/workflows/release.yml

```
@@ -1,347 +1,367 @@
-      - name: Log
-        run: |
-          echo "tag result:   ${{ needs.tag.outputs.TAG }}"
-          echo "pkg version:  ${{ needs.tag.outputs.PKG_VER }}"
-
-  #
-  #
-  #
-  upload-assets:
-    runs-on: ubuntu-24.04
-    needs: [create-release]
-    steps:
-
-      - name: Download Artifact
-        uses: actions/download-artifact@v4
-        with:
-          name: revive-wasm
-          path: resolc/
-
-      - name: Download Artifact
-        uses: actions/download-artifact@v4
-        with:
-          name: revive-linux
-          path: resolc/
-
-      - name: upload resolc
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.create-release.outputs.upload_url }}
-          asset_path: ./resolc/resolc
-          asset_name: resolc-static-linux
-          asset_content_type: application/octet-stream
-
-      - name: upload resolc.js
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.create-release.outputs.upload_url }}
-          asset_path: ./resolc/resolc.js
-          asset_name: resolc.js
-          asset_content_type: application/octet-stream
-
-      - name: upload resolc.wasm
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.create-release.outputs.upload_url }}
-          asset_path: ./resolc/resolc.wasm
-          asset_name: resolc.wasm
-          asset_content_type: application/octet-stream
-
-      - name: upload resolc_web.js
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.create-release.outputs.upload_url }}
-          asset_path: ./resolc/resolc_web.js
-          asset_name: resolc_web.js
-          asset_content_type: application/octet-stream
\ No newline at end of file
+name: Release
+run-name: Release ${{ github.ref_name }}
+on:
+  push:
+    branches:
+      - "main"
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+env:
+  #rust-musl-cross:x86_64-musl
+  RUST_MUSL_CROSS_IMAGE: messense/rust-musl-cross@sha256:68b86bc7cb2867259e6b233415a665ff4469c28b57763e78c3bfea1c68091561
+  RUST_LOG: trace
+
+jobs:
+  tag:
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: write
+    outputs:
+      TAG: ${{ steps.versions.outputs.TAG }}
+      PKG_VER: ${{ steps.versions.outputs.PKG_VER }}
+      RELEASE_NOTES: ${{ steps.versions.outputs.RELEASE_NOTES }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: "true"
+          fetch-depth: 0
+
+      - name: Versions
+        id: versions
+        run: |
+          export CURRENT_TAG=$(git describe --tags --abbrev=0 --exclude "llvm-*")
```

```suggestion
          export CURRENT_TAG=$(git describe --tags --abbrev=0 --exclude "llvm-*" --exclude "*-revive*")
```

there is no `llvm-` prefix for llvm releases now, so `release` will be triggered on every push after llvm tag
https://github.com/paritytech/revive/actions/runs/13454276537/job/37594743538#step:3:2

---

I think the proper fix is to make sure we retain the llvm tag: https://github.com/paritytech/revive/pull/224

---

Ah god dammit, the tag should've been `llvm-18.1.8-revive.6ad7908`...